app/com/linkedin/drelephant/exceptions/core/ExceptionFingerprintingRunner.java

```diff
@@ -99,20 +116,103 @@ private void saveDriverExceptionLogForExceptionFingerPrinting(List<ExceptionInfo
       sparkJobException.appId = _appResult.id;
       sparkJobException.taskId = NOT_APPLICABLE;
       sparkJobException.jobName = getJobName(_appResult.jobExecUrl);
-      if (driverExceptionInfo.isPresent()) {
-        sparkJobException.exceptionLog = driverExceptionInfo.get().getExcptionStackTrace();
+      sparkJobException.logSourceInformation = processLogSourceInformation(logSourceInformation);
+      //sparkJobException.exceptionLog = exceptionsTrace;
+      if (exceptionsTrace.trim().length() > 2) {
+        sparkJobException.exceptionLog = exceptionsTrace;
       } else {
         sparkJobException.exceptionLog = "Couldn't gather driver logs for the job";
       }
       sparkJobException.exceptionType = ExceptionInfo.ExceptionSource.DRIVER.toString();
-      sparkJobException.save();
+      saveSparkJobException(sparkJobException, exceptionInfoList);
       jobsExceptionFingerPrinting.save();
     }
   }
 
+  private void saveSparkJobException(JobsExceptionFingerPrinting sparkJobException,
+      List<ExceptionInfo> exceptionInfoList) {
+    try {
+      sparkJobException.save();
+      debugLog(" Final String to be saved in dB " + sparkJobException.exceptionLog);
+    } catch (PersistenceException pe) {
```

Comment:
What can be reason of failure here, which you are trying to resolve with reducing number of exception ?

Reply:
Data Size , currently exception log size is varchar(10000) , if the exceptions are beyond that size . Persistence exception comes , hence trying to reduce the data size , in order to store data .

Reply:
LogMessage should be constructed in a way that it is not more than 10000. Please remove this trial and error part.